via/services/youtube.py

```diff
@@ -11,6 +11,19 @@ def __init__(self, enabled: bool):
     def enabled(self):
         return self._enabled
 
+    def canonical_video_url(self, video_id: str) -> str:
+        """
+        Return the canonical URL for a YouTube video.
+
+        This is used as the URL which YouTube transcript annotations are
+        associated with.
+        """
+        if not video_id:
+            raise ValueError("Invalid video ID")
```

Comment:
I wouldn't bother with this `if`: the `quote_plus()` call below will raise `TypeError` if `video_id` is `None` (and perhaps also if `video_id` is invalid in other ways?). You could catch that `TypeError` and re-raise it as a `ValueError` if you want but given that nothing catches either I wouldn't bother.

We don't generally add validation code like this to our Python methods that're just called by other Python methods of ours. If you're worried about invalid video IDs coming from user-supplied data the place to validate that is at the edge: in a webargs schema on whatever views accept a `video_id` param, schema validation failures will trigger proper error responses

Reply:
My thinking here was to make it slightly easier to debug if an empty string gets passed. I'm fine with just removing this though.

Reply:
Yeah, I'd say that empty-string video IDs shouldn't be getting into the code in the first place. If they are then we're missing some validation on some view(s)

Reply:
OK, I have removed the `video_id` check here.